gpkit/constraints/single_equation.py

```diff
@@ -41,16 +41,25 @@ def latex(self, excluded=None):
         if not excluded:
             excluded = ["units"]  # previously bool(self.left.units)
         latex_oper = self.latex_opers[self.oper]
-        return ("%s %s %s" % (self.left.latex(excluded), latex_oper,
-                              self.right.latex(excluded)))
+        latexleft = trycall(self.left, "latex", excluded, str(self.left))
+        latexright = trycall(self.right, "latex", excluded, str(self.right))
+        return ("%s %s %s" % (latexleft, latex_oper, latexright))
 
     def sub(self, subs):
         "Returns a substituted version of this constraint."
-        subbed = self.func_opers[self.oper](self.left.sub(subs),
-                                            self.right.sub(subs))
+        subbedleft = trycall(self.left, "sub", subs, self.left)
+        subbedright = trycall(self.right, "sub", subs, self.right)
+        subbed = self.func_opers[self.oper](subbedleft, subbedright)
```

Comment:
Okay! Between this and another issue, I'm thinking the right solution might be to cast the left and right sides of a constraint to gpkit objects? (being signomials / nomial arrays of signomials) instead of staying 'raw'. Doing so would also fix this problem...